src/main/java9/org/jboss/modules/JDKSpecific.java

```diff
@@ -188,6 +191,10 @@ static void addInternalPackages(final List<String> list) {
         // none in Java 9+
     }
 
+    static <T> Iterable<T> findServices(final Class<T> serviceType, final Predicate<Class<?>> filter, final ClassLoader classLoader) {
+       return ServiceLoader.load(serviceType, classLoader).stream().filter(p -> filter.test(p.type())).map(ServiceLoader.Provider::get).collect(Collectors.toSet());
```

Comment:
Again don't return a collection here, instead use `.stream().iterator()` and then filter it like above.  I'm not sure how ServiceConfigurationErrors are handled in that case but it should be tested.

Reply:
We'll let the ServiceConfigurationErrors to be handled by users using this method.